lwr_hw/src/lwr_hw_fri_node.cpp

```diff
@@ -106,14 +107,17 @@ int main( int argc, char** argv )
   ros::Time last(ts.tv_sec, ts.tv_nsec), now(ts.tv_sec, ts.tv_nsec);
   ros::Duration period(1.0);
 
+  float sampling_time = lwr_robot.getSampleTime();
+  ROS_INFO("Sampling time on robot: %f", sampling_time);
+
   //the controller manager
   controller_manager::ControllerManager manager(&lwr_robot, lwr_nh);
 
   // run as fast as possible
   while( !g_quit )
   {
     // get the time / period
-    if (!clock_gettime(CLOCK_REALTIME, &ts))
+    if (!clock_gettime(CLOCK_MONOTONIC, &ts))
```

Comment:
Using `CLOCK_MONOTONIC` to set the current `time` in the call to `update` has caused some issues in other packages before (e.g. scheduling trajectories far into the future by the joint trajectory controller, publishing messages with timestamps incompatible with other ROS nodes). See conversation from [this comment](https://github.com/ThomasTimm/ur_modern_driver/issues/44#issuecomment-216267019).

Both the `ur_modern_driver` and the `ros_control_boilerplate` have settled on using the monotonic clock to compute the elapsed time, but setting the current time using `ros::Time::now()`.